app.rb

```diff
@@ -2,123 +2,166 @@
 require './student'
 require './book'
 require './teacher'
+require './save_data'
 
 class App
-  def show_books(books)
-    if books.empty?
+  attr_reader :book, :person, :rental
+
+  include DataSaver
+
+  def initialize
+    @book = show_books
+    @person = show_people
+    @rental = show_rentals(@book, @person)
+    @student_data = get_data('people')
+    @teach_data = get_data('people')
+    @book_data = get_data('books')
+    @rental_data = get_data('rental')
+  end
+
+  def list_books
+    if @book.empty?
       puts 'No books found in library'
     else
-      books.each do |book|
-        puts "title: #{book.title}, Auther: #{book.author}"
+      book.each do |item|
+        puts "Title: #{item.title}, Author: #{item.author}"
       end
     end
   end
 
-  def show_people(people)
-    if people.empty?
+  def list_people
+    if @person.empty?
       puts 'No person found'
     else
-      people.each do |person|
-        puts "Name: #{person.name}, Age: #{person.age},ID: #{person.id}"
+      person.each do |item|
+        puts "[#{item.class}] - Name: #{item.name}, Age: #{item.age}, ID: #{item.id}"
       end
     end
   end
 
-  def add_people(people)
-    print('press 1 to add a student or press 2 to add a teacher')
+  def handle_data(data, people)
+    case people
+    when 'student'
+      student_data_hash = { id: data.id, name: data.name, age: data.age, class: 'student' }
+      # $student_data = get_data('people')
+      @student_data.push(student_data_hash)
+    when 'teacher'
+      teacher_data_hash = { id: data.id, name: data.name, age: data.age, class: 'teacher' }
+      @teach_data.push(teacher_data_hash)
+    end
+  end
+
+  def add_people
+    puts 'press 1 to add a student or press 2 to add a teacher:'
     person_to_add = gets.chomp.to_i
-    print 'Age? '
+    puts 'Age: '
     age = gets.chomp.to_i
-    print 'Name? '
+    puts 'Name: '
     name = gets.chomp
 
     case person_to_add
     when 1
-      print 'Do you have parent Permission? [Y/N]:'
+      puts 'Do you have parent Permission? [Y/N]: '
       user_response = gets.chomp.capitalize
-      user_permission = true if user_response == 'Y'
-      user_permission = false if user_response == 'N'
-      people.push(Student.new(nil, age, name, parent_permission: user_permission))
+      user_permission = user_response == 'Y'
+      @person << people_data = Student.new(nil, age, name, parent_permission: user_permission)
+      handle_data(people_data, 'student')
```

Comment:
- [ ] Great work refactoring your code save upon exit. However, there is a bug. I noticed that whenever I create a **_student_** and exit the app. The student data is lost. So upon reentry, I can't view the student profile. Kindly fix this. 👍